### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 7

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#247] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315193-60c0e1fa-cc1a-4a86-987e-f520b857056e.png)

**Comment by LuginiAndrea** [comment]
Dimostriamo inanzitutto l'iniettività.
Ipotizziamo per assurdo che f non sia iniettiva, e quindi che date due coppie 
$$\exists a_1,a_2,b_1,b_2 \in Z \textrm{ t.c. } \left(a_1, b_1\right) \neq \left(a_2, b_2\right) \, \wedge \left(a_1+b_1,a_1+2b_1\right) = \left(a_2+b_2,a_2+2b_2\right)$$
Possiamo scrivere il sistema di equazioni
$$a_1+b_1=a_2+b_2$$
$$a_1+2b_1 = a_2 + 2b_2$$
Sottraendo le due equazioni otteniamo
$$-b_1 = -b_2 \Longrightarrow a_1 + b = a_2 + b \Longleftrightarrow a_1 = a_2$$
Siamo in contraddizione con la nostra ipotesi, quindi la funzione è iniettiva 

Discorso leggermente più complicato mostrare la suriettività. 
Prendiamo ora il sistrema di equazioni
$$a + b$$
$$a + 2b$$
Possiamo facilmente notare che la differenza fra le due è $= -b$.
Prendiamo ora due numeri $x, y \in Z$. In quanto per ipotesi $f$ è suriettiva
$$\left(x, y\right) = \left(a+b, a+2b\right)$$
Dal sistema di equazioni di prima sappiamo che 
$$x = y - b$$
Entrambe sono operazioni che garantiscono la suriettività, quindi ogni $x, y \in Z$ è raggiungibile dalla nostra funzione $f$.

Poichè abbiamo dimostrato che $f$ è iniettiva e suriettiva, possiamo concludere che $f$ è biettiva
